app/components/channel_drawer/channel_drawer.js

```diff
@@ -166,35 +170,62 @@ export default class ChannelDrawer extends PureComponent {
         const {
             actions,
             currentChannel,
+            currentDisplayName,
             currentTeam,
-            currentUserId
+            currentUserId,
+            intl
         } = this.props;
 
         const {
             handleSelectChannel,
             joinChannel,
             makeDirectChannel,
             markChannelAsRead,
+            setChannelDisplayName,
             setChannelLoading,
             viewChannel
         } = actions;
 
+        markChannelAsRead(currentChannel.id);
+        setChannelLoading();
+        viewChannel(currentChannel.id);
+        setChannelDisplayName(channel.display_name);
+
         if (channel.type === General.DM_CHANNEL) {
-            markChannelAsRead(currentChannel.id);
-            setChannelLoading();
-            viewChannel(currentChannel.id);
             this.closeChannelDrawer();
-            InteractionManager.runAfterInteractions(() => {
-                makeDirectChannel(channel.id);
+            InteractionManager.runAfterInteractions(async () => {
+                const result = await makeDirectChannel(channel.id);
+                if (!result) {
+                    setChannelDisplayName(currentDisplayName);
+                    Alert.alert(
+                        '',
+                        intl.formatMessage({
+                            id: 'mobile.open_dm.error',
+                            defaultMessage: "We couldn't open a direct message with {displayName}. Please check your connection and try again."
+                        }, {displayName: channel.display_name})
+                    );
+                }
             });
         } else {
-            markChannelAsRead(currentChannel.id);
-            setChannelLoading();
-            viewChannel(currentChannel.id);
-            joinChannel(currentUserId, currentTeam.id, channel.id);
+            let didJoin = false;
+            joinChannel(currentUserId, currentTeam.id, channel.id).then((result) => {
```

Comment:
There's a race condition here since we're not waiting for `joinChannel` to complete before closing the drawer and checking `didJoin`. We need to await the joinChannel or move the drawer closing into the .then block